app/views/layouts/application.html.erb

```diff
@@ -105,15 +105,9 @@
     <% end %>
 
     <%- if @preloaded.present? %>
-      <script>
-      (function() {
-        var ps = require('preload-store').default;
-        <%- @preloaded.each do |key, json| %>
-          ps.store("<%= key %>", <%= escape_unicode(json) %>);
-        <% end %>
-      })();
-      </script>
+      <div id="data-preloaded" data-preloaded="<%= @preloaded.transform_values(&method(:escape_unicode)).to_json %>"></div>
```

Comment:
@preloaded.transform_values(&method(:escape_unicode)).to_json is a lot to unwrap in your head 

I would go with a helper here called: `preloaded_json` and make sure there is a test case for it that demos how it works, personally I kind of prefer:

transform_values {|v| escape_unicode(v)} , I am pretty sure it would bench faster as well in a micro bench but the more important thing is that it is a clearer pattern